tools/generator.js

```diff
@@ -0,0 +1,163 @@
+const Lang = imports.lang;
+const System = imports.system;
+
+const Gio = imports.gi.Gio;
+
+const Generator = new Lang.Class({
```

Comment:
`Generator` is a built-in type in later JS versions, so best not to use this name.